### PR TITLE
[relay-compiler] relay 14 compiler types update

### PR DESCRIPTION
**Pull request by maraisr**
Myself and @levibuzolic will start more actively maintaining the relay suite of libraries. This here solidifies this in adding us as maintainers as well as bumping the version to support that of relay 14, the new Rust compiler.

### CODE REVIEW

**Comment by maraisr** [comment]
the broken tests will be fixed once https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61770 is merged and rebased into here.

**Review by levibuzolic — APPROVED** [review]

File: types/relay-compiler/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for relay-compiler 8.0
+// Type definitions for relay-compiler 14.0
 // Project: https://relay.dev
 // Definitions by: n1ru4l <https://github.com/n1ru4l>
 //                 Eloy Durán <https://github.com/alloy>
```
Comment (levibuzolic) [inline types/relay-compiler/index.d.ts:4]: Looks like we lost the added lines here
```suggestion
//                 Eloy Durán <https://github.com/alloy>
//                 Marais Rossouw <https://github.com/maraisr>
//                 Levi Buzolic <https://github.com/levibuzolic>
```

File: types/relay-compiler/tsconfig.json [hunk 2]
```diff
@@ -1,11 +1,6 @@
 {
     "compilerOptions": {
         "module": "commonjs",
```
Comment (levibuzolic) [inline types/relay-compiler/tsconfig.json:3]: I think we still need a `lib` here.

```suggestion
        "module": "commonjs",
        "lib": [
          "es6"
        ],
```

https://github.com/DefinitelyTyped/DefinitelyTyped/runs/7890150582?check_suite_focus=true#step:6:86

```
1> relay-compiler failing:
1> Error: Must specify "lib", usually to `"lib": ["es6"]` or `"lib": ["es6", "dom"]`.
1>     at checkTsconfig (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/checks.ts:117:11)
1>     at testTypesVersion (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:229:16)
1>     at runTests (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:197:7)
```

**Review by levibuzolic — APPROVED** [review]

**Review by levibuzolic — APPROVED** [review]

**Comment by maraisr** [comment]
ready to merge

**Comment by peterblazejewicz** [comment]
> the broken tests will be fixed once #61770 is merged and rebased into here.

what happened to those tests?

**Comment by maraisr** [comment]
The library on npm does nothing other than rerun a string. So wasn't sure if we needed a test for this? 

And yeah, fixed the types issue in that other pr.

File: types/relay-compiler/index.d.ts [hunk 3]
```diff
@@ -7,66 +7,6 @@
...
 // TypeScript Version: 3.4
 
-import * as ASTConvert from './lib/core/ASTConvert';
-import { CompilerContext } from './lib/core/CompilerContext';
-import * as Parser from './lib/core/RelayParser';
-import * as Printer from './lib/core/IRPrinter';
-import * as IRTransforms from './lib/core/RelayIRTransforms';
-import * as IRVisitor from './lib/core/IRVisitor';
-import * as SchemaUtils from './lib/core/GraphQLSchemaUtils';
+declare const binary: string;
```
Comment (peterblazejewicz) [inline types/relay-compiler/index.d.ts:10]:
```suggestion
declare const binary: string | null;
```
https://github.com/facebook/relay/blob/main/packages/relay-compiler/index.js#L30
so the test should expect symbol to be"
```ts
// $ExpectType string | null
```

File: types/relay-compiler/tsconfig.json [hunk 4]
```diff
@@ -1,11 +1,7 @@
...
-        "lib": [
-            "dom",
-            "es6",
-            "esnext.asynciterable"
-        ],
+        "lib": ["ES6"],
```
Comment (jakebailey) [inline types/relay-compiler/tsconfig.json:4]: I think that if you convert this to `"es6"`, the bot shouldn't complain about the config anymore.

**Review by levibuzolic — APPROVED** [review]

**Comment by levibuzolic** [comment]
Ready to merge

**Pull request merged**